test/index.test.js

```diff
@@ -32,299 +35,386 @@ Object.defineProperty(loaderContextMock, "options", {
-                // check for file excerpt
-                err.message.should.match(/Property "some-value" must be followed by a ':'/);
-                err.message.should.match(/\(line 2, column 5\)/);
-                err.message.indexOf(pathToErrorFile).should.not.equal(-1);
-                done();
-            });
-        });
-        it("should output understandable errors when a file could not be found", (done) => {
-            runWebpack({
-                entry: pathToSassLoader + "!" + pathToErrorFileNotFound
-            }, (err) => {
-                err.message.should.match(/@import "does-not-exist";/);
-                err.message.should.match(/File to import not found or unreadable: does-not-exist/);
-                err.message.should.match(/\(line 1, column 1\)/);
-                err.message.indexOf(pathToErrorFileNotFound).should.not.equal(-1);
-                done();
-            });
-        });
-        it("should not auto-resolve imports with explicit file names", (done) => {
-            runWebpack({
-                entry: pathToSassLoader + "!" + pathToErrorFileNotFound2
-            }, (err) => {
-                err.message.should.match(/@import "\.\/another\/_module\.scss";/);
-                err.message.should.match(/File to import not found or unreadable: \.\/another\/_module\.scss/);
-                err.message.should.match(/\(line 1, column 1\)/);
-                err.message.indexOf(pathToErrorFileNotFound2).should.not.equal(-1);
-                done();
-            });
-        });
-        it("should not swallow errors when trying to load node-sass", (done) => {
-            mockRequire.reRequire(pathToSassLoader);
-            const module = require("module");
-            const originalResolve = module._resolveFilename;
+            describe("errors", () => {
+                it("should throw an error in synchronous loader environments", () => {
+                    try {
+                        sassLoader.call(Object.create(loaderContextMock), "");
+                    } catch (err) {
+                        // check for file excerpt
+                        err.message.should.equal("Synchronous compilation is not supported anymore. See https://github.com/webpack-contrib/sass-loader/issues/333");
+                    }
+                });
+                it("should output understandable errors in entry files", (done) => {
+                    runWebpack({
+                        entry: pathToErrorFile
+                    }, {}, (err) => {
+                        if (implementation === nodeSass) {
+                            err.message.should.match(/Property "some-value" must be followed by a ':'/);
+                            err.message.should.match(/\(line 2, column 5\)/);
+                        } else {
+                            err.message.should.match(/Expected "{"./);
+                            err.message.should.match(/\(line 3, column 1\)/);
+                        }
```

Comment:
In the long run it would probably be better to compare these errors with errors that have been snapshotted during `createSpec`, but that's ok for now I guess.